File: OsmAnd-java/src/main/java/net/osmand/IndexConstants.java
package net.osmand;


public class IndexConstants {

	// Important : Every time you change schema of db upgrade version!!! 
	// If you want that new application support old index : put upgrade code in android app ResourceManager
	public static final int POI_TABLE_VERSION = 2;
	public static final int BINARY_MAP_VERSION = 2; // starts with 1
	public static final int VOICE_VERSION = 0; //supported download versions
	public static final int TTSVOICE_VERSION = 1; //supported download versions

	public static final String SQLITE_EXT = ".sqlitedb";
	public static final String TEMP_SOURCE_TO_LOAD = "temp";

	public static final String POI_INDEX_EXT = ".poi.odb";

	public static final String ZIP_EXT = ".zip";
	public static final String BINARY_MAP_INDEX_EXT = ".obf";
	public static final String BINARY_MAP_INDEX_EXT_ZIP = ".obf.zip";

	public static final String BINARY_WIKIVOYAGE_MAP_INDEX_EXT = ".sqlite";
	public static final String BINARY_TRAVEL_GUIDE_MAP_INDEX_EXT = ".travel.obf";
	public static final String BINARY_TRAVEL_GUIDE_MAP_INDEX_EXT_ZIP = ".travel.obf.zip";
	public static final String BINARY_WIKI_MAP_INDEX_EXT = ".wiki.obf";
	public static final String BINARY_WIKI_MAP_INDEX_EXT_ZIP = ".wiki.obf.zip";
	public static final String BINARY_ROAD_MAP_INDEX_EXT = ".road.obf";
	public static final String BINARY_ROAD_MAP_INDEX_EXT_ZIP = ".road.obf.zip";
	public static final String BINARY_SRTM_MAP_INDEX_EXT = ".srtm.obf";
	public static final String BINARY_SRTM_FEET_MAP_INDEX_EXT = ".srtmf.obf";
	public static final String BINARY_SRTM_MAP_INDEX_EXT_ZIP = ".srtm.obf.zip";
	public static final String BINARY_SRTM_FEET_MAP_INDEX_EXT_ZIP = ".srtmf.obf.zip";
	public static final String BINARY_DEPTH_MAP_INDEX_EXT = ".depth.obf";
	public static final String BINARY_DEPTH_MAP_INDEX_EXT_ZIP = ".depth.obf.zip";
	public static final String EXTRA_EXT = ".extra";
	public static final String EXTRA_ZIP_EXT = ".extra.zip";
	public static final String OSM_GZ_EXT = ".osm.gz";
	public static final String HTML_EXT = ".html";
	public static final String GEN_LOG_EXT = ".gen.log";
	public static final String DOWNLOAD_EXT = ".download";
	public static final String TIF_EXT = ".tif";
	public static final String TIFF_DB_EXT = ".tiff.db";
	public static final String WEATHER_EXT = ".tifsqlite";
	public static final String WEATHER_MAP_INDEX_EXT = ".tifsqlite.zip";

	public static final String VOICE_INDEX_EXT_ZIP = ".voice.zip";
	public static final String TTSVOICE_INDEX_EXT_JS = "tts.js";
	public static final String ANYVOICE_INDEX_EXT_ZIP = "voice.zip";  //to cactch both voices, .voice.zip and .ttsvoice.zip

	public static final String FONT_INDEX_EXT = ".otf";
	public static final String FONT_INDEX_EXT_ZIP = ".otf.zip";

	public static final String OSMAND_SETTINGS_FILE_EXT = ".osf";

	public static final String ROUTING_FILE_EXT = ".xml";

	public static final String RENDERER_INDEX_EXT = ".render.xml";
	public static final String ADDON_RENDERER_INDEX_EXT = ".addon" + RENDERER_INDEX_EXT;

	public static final String GPX_FILE_EXT = ".gpx";
	public static final String GPX_GZ_FILE_EXT = ".gpx.gz";

	public static final String WPT_CHART_FILE_EXT = ".wpt.chart";
	public static final String SQLITE_CHART_FILE_EXT = ".3d.chart";
	public static final String HELP_ARTICLE_FILE_EXT = ".mht";

	public static final String AVOID_ROADS_FILE_EXT = ".geojson";

	public static final String OBJ_FILE_EXT = ".obj";
	public static final String TXT_EXT = ".txt";

	public static final String POI_TABLE = "poi";

	public static final String INDEX_DOWNLOAD_DOMAIN = "download.osmand.net";
	public static final String APP_DIR = "osmand/";
	public static final String MAPS_PATH = "";
	public static final String HIDDEN_DIR = "hidden/";
	public static final String BACKUP_INDEX_DIR = "backup/";
	public static final String HIDDEN_BACKUP_DIR = HIDDEN_DIR + BACKUP_INDEX_DIR;
	public static final String GPX_INDEX_DIR = "tracks/";
	public static final String FAVORITES_INDEX_DIR = "favorites/";
	public static final String MAP_MARKERS_INDEX_DIR = "/map markers";
	public static final String GPX_RECORDED_INDEX_DIR = GPX_INDEX_DIR + "rec/";
	public static final String GPX_IMPORT_DIR = GPX_INDEX_DIR + "import/";
	public static final String TILES_INDEX_DIR = "tiles/";
	public static final String LIVE_INDEX_DIR = "live/";
	public static final String TOURS_INDEX_DIR = "tours/";
	public static final String SRTM_INDEX_DIR = "srtm/";
	public static final String NAUTICAL_INDEX_DIR = "nautical/";
	public static final String ROADS_INDEX_DIR = "roads/";
	public static final String WIKI_INDEX_DIR = "wiki/";
	public static final String HELP_INDEX_DIR = "help/";
	public static final String ARTICLES_DIR = HELP_INDEX_DIR + "articles/";
	public static final String WIKIVOYAGE_INDEX_DIR = "travel/";
	public static final String GPX_TRAVEL_DIR = GPX_INDEX_DIR + WIKIVOYAGE_INDEX_DIR;
	public static final String AV_INDEX_DIR = "avnotes/";
	public static final String FONT_INDEX_DIR = "fonts/";
	public static final String VOICE_INDEX_DIR = "voice/";
	public static final String RENDERERS_DIR = "rendering/";
	public static final String ROUTING_XML_FILE = "routing.xml";
	public static final String SETTINGS_DIR = "settings/";
	public static final String TEMP_DIR = "temp/";
	public static final String ROUTING_PROFILES_DIR = "routing/";
	public static final String PLUGINS_DIR = "plugins/";
	public static final String GEOTIFF_SQLITE_CACHE_DIR = "geotiff_sqlite_cache/";
	public static final String GEOTIFF_DIR = "geotiff/";
	public static final String CLR_PALETTE_DIR = "color-palette/";
	public static final String WEATHER_FORECAST_DIR = "weather_forecast/";
	public static final String MODEL_3D_DIR = "models/";
	public static final String OPENGL_SHADERS_CACHE_DIR = "opengl_shaders_cache/";

	public static final String VOICE_PROVIDER_SUFFIX = "-tts";
	public static final String MODEL_NAME_PREFIX = "model_";
	public static final String COLOR_PALETTE_DIR = "color-palette/";
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/IndexConstants.kt
package net.osmand.shared

object IndexConstants {

	const val SQLITE_EXT = ".sqlitedb"

	const val POI_INDEX_EXT = ".poi.odb"

	const val ZIP_EXT = ".zip"
	const val BINARY_MAP_INDEX_EXT = ".obf"
	const val BINARY_MAP_INDEX_EXT_ZIP = ".obf.zip"

	const val BINARY_WIKIVOYAGE_MAP_INDEX_EXT = ".sqlite"
	const val BINARY_TRAVEL_GUIDE_MAP_INDEX_EXT = ".travel.obf"
	const val BINARY_TRAVEL_GUIDE_MAP_INDEX_EXT_ZIP = ".travel.obf.zip"
	const val BINARY_WIKI_MAP_INDEX_EXT = ".wiki.obf"
	const val BINARY_WIKI_MAP_INDEX_EXT_ZIP = ".wiki.obf.zip"
	const val BINARY_ROAD_MAP_INDEX_EXT = ".road.obf"
	const val BINARY_ROAD_MAP_INDEX_EXT_ZIP = ".road.obf.zip"
	const val BINARY_SRTM_MAP_INDEX_EXT = ".srtm.obf"
	const val BINARY_SRTM_FEET_MAP_INDEX_EXT = ".srtmf.obf"
	const val BINARY_SRTM_MAP_INDEX_EXT_ZIP = ".srtm.obf.zip"
	const val BINARY_SRTM_FEET_MAP_INDEX_EXT_ZIP = ".srtmf.obf.zip"
	const val BINARY_DEPTH_MAP_INDEX_EXT = ".depth.obf"
	const val BINARY_DEPTH_MAP_INDEX_EXT_ZIP = ".depth.obf.zip"
	const val EXTRA_EXT = ".extra"
	const val EXTRA_ZIP_EXT = ".extra.zip"
	const val OSM_GZ_EXT = ".osm.gz"
	const val HTML_EXT = ".html"
	const val GEN_LOG_EXT = ".gen.log"
	const val DOWNLOAD_EXT = ".download"
	const val TIF_EXT = ".tif"
	const val TIFF_DB_EXT = ".tiff.db"
	const val WEATHER_EXT = ".tifsqlite"
	const val WEATHER_MAP_INDEX_EXT = ".tifsqlite.zip"

	const val VOICE_INDEX_EXT_ZIP = ".voice.zip"
	const val TTSVOICE_INDEX_EXT_JS = "tts.js"
	const val ANYVOICE_INDEX_EXT_ZIP = "voice.zip" // to catch both voices, .voice.zip and .ttsvoice.zip

	const val FONT_INDEX_EXT = ".otf"
	const val FONT_INDEX_EXT_ZIP = ".otf.zip"

	const val OSMAND_SETTINGS_FILE_EXT = ".osf"

	const val ROUTING_FILE_EXT = ".xml"

	const val RENDERER_INDEX_EXT = ".render.xml"
	const val ADDON_RENDERER_INDEX_EXT = ".addon$RENDERER_INDEX_EXT"

	const val GPX_FILE_EXT = ".gpx"
	const val GPX_GZ_FILE_EXT = ".gpx.gz"

	const val WPT_CHART_FILE_EXT = ".wpt.chart"
	const val SQLITE_CHART_FILE_EXT = ".3d.chart"
	const val HELP_ARTICLE_FILE_EXT = ".mht"

	const val AVOID_ROADS_FILE_EXT = ".geojson"

	const val OBJ_FILE_EXT = ".obj"

	const val POI_TABLE = "poi"

	const val INDEX_DOWNLOAD_DOMAIN = "download.osmand.net"
	const val APP_DIR = "osmand/"
	const val MAPS_PATH = ""
	const val HIDDEN_DIR = "hidden/"
	const val BACKUP_INDEX_DIR = "backup/"
	const val HIDDEN_BACKUP_DIR = HIDDEN_DIR + BACKUP_INDEX_DIR
	const val GPX_INDEX_DIR = "tracks/"
	const val FAVORITES_INDEX_DIR = "favorites/"
	const val MAP_MARKERS_INDEX_DIR = "/map markers"
	const val GPX_RECORDED_INDEX_DIR = GPX_INDEX_DIR + "rec/"
	const val GPX_IMPORT_DIR = GPX_INDEX_DIR + "import/"
	const val TILES_INDEX_DIR = "tiles/"
	const val LIVE_INDEX_DIR = "live/"
	const val TOURS_INDEX_DIR = "tours/"
	const val SRTM_INDEX_DIR = "srtm/"
	const val NAUTICAL_INDEX_DIR = "nautical/"
	const val ROADS_INDEX_DIR = "roads/"
	const val WIKI_INDEX_DIR = "wiki/"
	const val HELP_INDEX_DIR = "help/"
	const val ARTICLES_DIR = HELP_INDEX_DIR + "articles/"
	const val WIKIVOYAGE_INDEX_DIR = "travel/"
	const val GPX_TRAVEL_DIR = GPX_INDEX_DIR + WIKIVOYAGE_INDEX_DIR
	const val AV_INDEX_DIR = "avnotes/"
	const val FONT_INDEX_DIR = "fonts/"
	const val VOICE_INDEX_DIR = "voice/"
	const val RENDERERS_DIR = "rendering/"
	const val ROUTING_XML_FILE = "routing.xml"
	const val SETTINGS_DIR = "settings/"
	const val TEMP_DIR = "temp/"
	const val ROUTING_PROFILES_DIR = "routing/"
	const val PLUGINS_DIR = "plugins/"
	const val GEOTIFF_SQLITE_CACHE_DIR = "geotiff_sqlite_cache/"
	const val GEOTIFF_DIR = "geotiff/"
	const val CLR_PALETTE_DIR = "color-palette/"
	const val WEATHER_FORECAST_DIR = "weather_forecast/"
	const val MODEL_3D_DIR = "models/"

	const val VOICE_PROVIDER_SUFFIX = "-tts"
	const val MODEL_NAME_PREFIX = "model_"
	const val COLOR_PALETTE_DIR = "color-palette/"


}
